src/analysis/convex/simplicial_complex/basic.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import analysis.convex.topology
+
+/-!
+# Simplicial complexes
+
+In this file, we define simplicial complexes in `𝕜`-modules. A simplicial complex is a collection
+of simplices closed by inclusion (of vertices) and intersection (of underlying sets).
+
+We model them by a downward-closed set of affine independent finite sets whose convex hulls "glue
+nicely", each finite set and its convex hull corresponding respectively to the vertices and the
+underlying set of a simplex.
+
+## Main declarations
+
+* `simplicial_complex 𝕜 E`: A simplicial complex in the `𝕜`-module `E`.
+* `simplicial_complex.vertices`: The zero dimensional faces of a simplicial complex.
+* `simplicial_complex.facets`: The maximal faces of a simplicial complex.
+
+## TODO
+
+Simplicial complexes can be generalized to affine spaces once `convex_hull` has been ported.
+-/
+
+open finset set
+
+variables {𝕜 E ι : Type*} [ordered_ring 𝕜] [add_comm_group E] [module 𝕜 E]
+
+variables (𝕜 E)
+
+/-- A simplicial complex in a `𝕜`-module is a collection of simplices which glue nicely together. -/
```

Comment:
What's the usual statement? Is it this?
```
∀ s t ∈ K.faces, ∃ u ∈ K.faces, convex_hull 𝕜 ↑s ∩ convex_hull 𝕜 ↑t = convex_hull 𝕜 u
```
because in that case `u` can simply be taken to be `s ∩ t` and the existential is just an added layer of pain.

Reply:
I think what you really want anyways to compare to other definitions is the reverse implication, i.e., if you're trying to prove something is a simplicial complex and you know the intersection of two simplices is a simplex, it has to be the simplex whose vertices are the intersection of the vertices of the two simplices. So adding a lemma about `simplicial_complex` doesn't help, you would want to add another constructor.

I'm not sure if that's necessary; I think it would be fine to just explain this in the docstring.
(In general, if there are multiple well-known equivalent ways to state a definition, I think we can just pick one and we don't necessarily have to also prove the equivalence to the other ways unless we need it for some internal purpose.)
Maybe also mention that the formulation used here includes "intersection of faces is a face *or empty*" automatically.

I would be inclined to use the statement of `convex_hull_inter_convex_hull` as the definition of the "meets nicely" condition, since it's more obviously the correct definition and I think you would basically never want to check this "by hand" anyways, but rather apply some general fact about convexity, which would presumably already be stated in the better `=` form. But it doesn't make a big difference.

Reply:
What about now?